receiver/receiver.go

```diff
@@ -20,6 +20,29 @@ import (
 	_ "github.com/open-telemetry/opentelemetry-service/internal/compression/grpc" // load in supported grpc compression encodings
 )
 
+// Host represents the entity where the receiver is being hosted. It is used to
+// allow communication between the receiver and its host.
+type Host interface {
+	// Context returns a context provided by the host to be used on the receiver
+	// operations.
+	Context() context.Context
+
+	// AsyncErrorChannel returns a channel used by a receiver to notify its
+	// host about any fatal error (that requires re-starting the receiver) that
```

Comment:
>that requires re-starting the receiver

Do we actually re-start receives on errors? From what I remember we just shutdown the entire service if an error is reported on on async channel.

Reply:
Any failure reported here will trigger a service shutdown but we could opt for something different later, errors reported here indicate that the receiver hit an unrecoverable error. I will re-word this to avoid confusion.